src/python/AsyncStageOut/TransferWorker.py

```diff
@@ -193,10 +198,13 @@ def files_for_transfer(self):
         ftscp copyjob per source:destination.
         """
         source_dests = self.source_destinations_by_user()
-        jobs = {}
-        jobs_lfn = {}
-        jobs_pfn = {}
-        jobs_report = {}
+        # Generate dictionary of new jobs.
+        # output is for big files, output_sm is for small files
+        output = {"jobs": {},
+                  "jobs_lfn": {},
+                  "jobs_pfn": {},
+                  "jobs_report": {}}
+        output_sm = output
```

Comment:
Maybe you do not need 2 dict. You could use only one and add the flag use_small_queue into this dict. In this way in addition to manage 1 dict instead of 2 you will be also able to move the duplication of code in lines 273 - 284 and 169 - 174.